src/message.ts

```diff
@@ -0,0 +1,93 @@
+import { IAddress, IMessage } from "./interface";
+import { DEFAULT_MESSAGE_VERSION, MESSAGE_PREFIX } from "./constants";
+import { Address } from "./address";
+
+const createKeccakHash = require("keccak");
+
+export class Message {
+    /**
+     * Actual message being signed.
+     */
+    public data: Uint8Array;
+    /**
+     * The message signature.
+     */
+    public signature?: Uint8Array;
+    /**
+     * Address of the wallet that performed the signing operation.
+     */
+    public address?: IAddress;
+    /**
+     * Number representing the message version.
+     */
+    public version?: number;
+
+    constructor(options: { data: Uint8Array; signature?: Uint8Array; address?: IAddress; version?: number }) {
+        this.data = options.data;
+        this.signature = options.signature;
+        this.address = options.address;
+        this.version = options.version ? options.version : DEFAULT_MESSAGE_VERSION;
```

Comment:
Somehow unexpected when version will be set to `0` (but all right in practice, since there is no version `0`). Maybe check for `undefined` instead of `options.version`?

(optional)

Reply:
As discussed, since `version = 0` it's not valid it ok how it is.

Reply:
`this.version = options.version || DEFAULT_MESSAGE_VERSION` would be shorter
(optional)

Reply:
Changed